core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -3318,7 +3318,18 @@ object Stream extends StreamLowPriority {
     }
   }
 
-  private[fs2] final class PartiallyAppliedFromBlockingIterator[F[_]](
+  object PartiallyAppliedFromIterator {
+    @deprecated("bincompat", "3.2.10")
+    private[fs2] def `apply$extension`[F[_], A](
+        dummy: Boolean,
+        iterator: Iterator[A],
+        chunkSize: Int,
+        F: Sync[F]
+    ): Stream[F, A] =
+      new PartiallyAppliedFromIterator(dummy).apply(iterator, chunkSize)(F)
+  }
```

Comment:
On Scala 2.12, adding an overload of a method defined on an `AnyVal` causes the the binary name of the original method to change, breaking bincompat. Fortunately, we can restore a method with the same signature manually.